src/sonic-yang-models/yang-models/sonic-device_metadata.yang

```diff
@@ -89,6 +89,12 @@ module sonic-device_metadata {
                         pattern "ToRRouter|LeafRouter|SpineChassisFrontendRouter|ChassisBackendRouter|ASIC";
                     }
                 }
+
+                leaf buffer_model {
+                    type string {
+                        pattern "dynamic|traditional";
+                    }
```

Comment:
please add description for this leaf to describe the function name.

Reply:
Hi @lguohan,
This leaf is added for dynamic buffer calculation.
The `dynamic` model represents the model in which the buffer configurations, like the headroom sizes and buffer pool sizes, are dynamically calculated based on the ports' speed, cable length, and MTU. This model is used by Mellanox so far.
The `traditional` model represents the model in which all the buffer configurations are statically configured in `CONFIG_DB` tables. This is the default model used by all other vendors.

Reply:
yes, can you add description field in the leaf node?

Reply:
Done